src/porepy/applications/convergence_analysis.py

```diff
@@ -88,22 +83,29 @@ def __init__(
         model_class,
         model_params: dict,
         levels: int = 1,
-        in_space: bool = True,
-        in_time: bool = False,
         spatial_refinement_rate: int = 1,
         temporal_refinement_rate: int = 1,
     ):
         # Sanity checks
-        if not in_space and not in_time:
-            raise ValueError("At least one type of analysis should be performed.")
+        if spatial_refinement_rate < 1 or temporal_refinement_rate < 1:
+            raise ValueError("Refinement rate cannot be less than 1.")
 
-        if not in_space and spatial_refinement_rate > 1:
-            warnings.warn("'spatial_refinement_rate' is not being used.")
-            spatial_refinement_rate = 1
+        if spatial_refinement_rate == 1 and temporal_refinement_rate == 1:
+            warnings.warn("No refinement (in space or time) will be performed.")
 
-        if not in_time and temporal_refinement_rate > 1:
-            warnings.warn("'temporal_refinement_rate' is not being used.")
-            temporal_refinement_rate = 1
+        # Check if analysis in space should be performed
+        if spatial_refinement_rate == 1:
+            self._in_space: bool = False
```

Comment:
It seems you're only using _in_time in one place and _in_space not at all. Are you sure they're needed?

Reply:
Yes, my bad. `self._in_time` and `self._in_space` have been fully purged now.